orttraining/orttraining/python/training/torchdynamo/ort_backend.py

```diff
@@ -0,0 +1,519 @@
+            # ORT always returns a tuple of outputs. If the original is a tuple, just returning
+            # ORT output is ok.
+            _nvtx_range_push("run_onnx_session_with_ortvaluevector")
+            onnx_outputs = _run_onnx_session_with_ortvaluevector(
+                onnx_session, input_names, args, input_devices, output_names, prim_outputs, output_devices
+            )
+            _nvtx_range_pop()
+            if self.ort_execution_info.assert_allclose_to_baseline:
+                # Compute baseline.
+                baseline_outputs = torch._prims.executor.execute(graph_module, *args, executor="aten")
+                # Ensure every output tensor is close to the corresponding baseline.
+                for onnx_output, baseline_output in zip(onnx_outputs, baseline_outputs):
+                    _assert_allclose_with_detailed_error_message(onnx_output, baseline_output)
+            return onnx_outputs
+        else:
+            assert isinstance(prim_outputs, torch.Tensor)
+            # ORT always returns a tuple of outputs. If the original output is a tensor,
+            # ORT output's first element must be extracted and returned. Otherwise, type
+            # mismatch may happen in downstream computation.
+            onnx_outputs = _run_onnx_session_with_ortvaluevector(
+                onnx_session, input_names, args, input_devices, output_names, (prim_outputs,), output_devices
+            )
+            assert len(onnx_outputs) == 1
+            if self.ort_execution_info.assert_allclose_to_baseline:
+                # Compute baseline.
+                baseline_outputs = torch._prims.executor.execute(graph_module, *args, executor="aten")
+                # Ensure output tensor is close to the corresponding baseline.
+                _assert_allclose_with_detailed_error_message(onnx_outputs[0], baseline_outputs)
+            return onnx_outputs[0]
+
+    def compile(self, graph_module: torch.fx.GraphModule, args) -> torch.fx.GraphModule:
+        # FX graph based partitioning based on ONNX supported ops.
+        if graph_module in self.partitioner_cache:
+            partitioned_prim_graph_module = self.partitioner_cache[graph_module]
+        else:
+            prim_graph_module = make_fx(graph_module, decomposition_table=aten2aten_decomp)(*args)
+            partitioner = CapabilityBasedPartitioner(
+                prim_graph_module, self.supported_ops, allows_single_node_partition=False
+            )
+            partitioned_prim_graph_module = partitioner.partition_and_fuse()
+            self.partitioner_cache[graph_module] = partitioned_prim_graph_module
+
+            # Overriding fused_module's __call__() function with ort_acclerated_call()
+            for node in partitioned_prim_graph_module.graph.nodes:
+                # TODO: use a better way to identify fused submodule
+                if node.op == "call_module" and "fused_" in node.name:
```

Comment:
Does it need to start with fused_, or is it fine having fused_ anywhere in the name? Would startswith be more efficient?

Reply:
It's copied from another official PyTorch backend. Let me keep it so I don't have to spend much time on a hack.